tests/factories/requests_.py

```diff
@@ -0,0 +1,50 @@
+import json
+from io import BytesIO
+
+import requests
+from factory import Factory, Faker
+
+
+class Response(Factory):
+    class Meta:
+        model = requests.Response
+
+    raw = None
+    json_data = None
+    encoding = "utf-8"
+    status_code = Faker(
+        "random_element", elements=[200, 201, 301, 304, 401, 404, 500, 501],
+    )
+    headers = None
+
+    @classmethod
+    def _adjust_kwargs(cls, **kwargs):
+        headers = kwargs["headers"] or {}
+        encoding = kwargs["encoding"]
+        raw = kwargs["raw"]
+        json_data = kwargs["json_data"]
+
+        if raw is None and json_data is not None:
+            raw = json.dumps(json_data)
+
+            if not headers:
+                headers["Content-Type"] = f"application/json; charset={encoding}"
+
+        kwargs["raw"] = BytesIO(raw.encode(encoding)) if isinstance(raw, str) else raw
+
+        # Requests seems to store these lower case and expects them that way
+        kwargs["headers"] = {key.lower(): value for key, value in headers.items()}
+
+        return kwargs
+
+    @classmethod
+    def _create(cls, model_class, *_args, **kwargs):
```

Comment:
Factoryboy kind of sucks for objects which aren't initialised using their constructor.

```python
Object(value=blah)   # All good!

# vs.

obj = Object()    # Do it yourself
obj.value = blah 
```

I tried a whole bunch of ways around this, but couldn't find any. There are `params` which are not directly mapped to constructor `kwargs` but I can't see any way of accessing them other than to define attributes, which doesn't help. So here we are.

Reply:
Yeah, I'm not 100% but AFAIK the `_create()` method you have here is the right thing to do / is what you have to do.

Factory Boy's default `_create()` method passes all the generated values to the model's `__init__()` as kwargs. You can use `inline_args` to get it to pass some of them as pargs instead of kwargs if necessary. But you can't tell it to set some of them using `setattr` post-creation.

I don't think Factory Boy sucks here though. It provides a default `_create()` method that works for most objects, and if you need to create your object in a different way you override `_create()` with your own :man_shrugging: If you weren't using Factory Boy you would still need to create a `Response` and then set attrs on it so you would need to write the equivalent of this `_create()` code anyway.